atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -149,6 +156,47 @@ def get_buildroot(self, worker_metadatas):
 
         return buildroots
 
+    def get_logs(self):
+        """
+        Build list of log files
+
+        :return: list, of log files
+        """
+
+        logs = None
+        output = []
+
+        # Collect logs from server
+        try:
+            logs = self.osbs.get_orchestrator_build_logs(self.build_id)
+        except OsbsException as ex:
+            self.log.error("unable to get build logs: %r", ex)
+            return output
+        except TypeError:
+            # Older osbs-client has no get_orchestrator_build_logs
+            self.log.error("OSBS client does not support get_orchestrator_build_logs")
+            return output
+
+        platform_logs = {}
+        for entry in logs:
+            platform = entry.platform
+            platform_logs.setdefault(platform, [])
+            if platform not in platform_logs.items():
```

Comment:
When will `platform` ever be in `platform_logs.items()`?

Should this be something like this:

```python
platform = entry.platform
# platform_logs.setdefault(platform, [])  # Don't use setdefault if we need to check if key exists
if platform not in platform_logs:  # This actually checks if platform is in keys of platform_logs, not items
  platorm_logs[plaform] = NamedTemporaryFile(...)
  
```

Reply:
Good catch.

Reply:
Ah, I forgot about something here when I ran into it last night. With just 'if platform not in platform_logs:', the log file was never getting created, for whatever reason -- the self-tests would bomb out on the following .write() call, saying that you can't use .write on a dict, iirc, so something wasn't quite working, but it seems .items() wasn't the right fix either. Prior code using a separate list of platforms did seem to work correctly, though I guess better understanding why the code as suggested by both @twaugh and @lcarva isn't behaving as expected.

Reply:
Original discussion: https://github.com/projectatomic/atomic-reactor/pull/782#discussion_r135921167

Reply:
It would help if I'd read @lcarva 's comment closer, it shows that the check would never be true, since the setdefault call right before it made sure that was never true.